src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,339 @@
+####################################
+# Implicit Adaptive Adam-Bashforth solvers
+####################################
+# (Hairer & Wanner 1996, Vol I, p.356-360, p.398-400, p.421-423)
+# ODE_MS Fixed-step, fixed-order multi-step numerical method
+# with Adams-Bashforth-Moulton coefficients
+function ode_ab_adaptive(F, y0, tspan, order::Integer=4;adaptive_order = true,
+                                                      max_order = 13,
+                                                      reltol = 1.0e-5,
+                                                      abstol = 1.0e-8,
+                                                      minstep=abs(tspan[1] - tspan[end])/1e12,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.5,
+                                                      facmin = .1,
+                                                      fac = .8,
+                                                      norm=Base.norm,
+                                                      points =:all)
+
+    # initialization of time span, which will have variable length
+    dof = length(y0)
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = [tstart]
+
+    # initialzation of y[n], which will have variable length as well
+    y = typeof(y0)[]
```

Comment:
`y = [deepcopy(y0)]` would do.  But if you use `make_consitent_types` then it will be more verbose, like now.